protocol/x/subaccounts/keeper/transfer_test.go

```diff
@@ -574,6 +574,599 @@ func TestWithdrawFundsFromSubaccountToAccount_DepositFundsFromAccountToSubaccoun
+				&constants.PerpetualPosition_OneISOShort,
+			},
+			recipientAssetPositions: keepertest.CreateUsdcAssetPosition(big.NewInt(600)),
+			recipientPerpetualPositions: []*types.PerpetualPosition{
+				&constants.PerpetualPosition_OneISOShort,
+			},
+			senderCollateralPoolBalance:    big.NewInt(1100),
+			recipientCollateralPoolBalance: big.NewInt(1100),
+			senderCollateralPoolAddr: authtypes.NewModuleAddress(
+				types.ModuleName + ":" + lib.UintToString(constants.PerpetualPosition_OneISOShort.PerpetualId),
+			),
+			recipientCollateralPoolAddr: authtypes.NewModuleAddress(
+				types.ModuleName + ":" + lib.UintToString(constants.PerpetualPosition_OneISOShort.PerpetualId),
+			),
+			quantums:    big.NewInt(500),
+			expectedErr: types.ErrFailedToUpdateSubaccounts,
+		},
+		"Send between isolated subaccounts (different perp), sender does not have enough balance": {
+			asset:                *constants.Usdc,
+			senderAssetPositions: keepertest.CreateUsdcAssetPosition(big.NewInt(100)),
+			senderPerpetualPositions: []*types.PerpetualPosition{
+				&constants.PerpetualPosition_OneISOShort,
+			},
+			recipientAssetPositions: keepertest.CreateUsdcAssetPosition(big.NewInt(600)),
+			recipientPerpetualPositions: []*types.PerpetualPosition{
+				&constants.PerpetualPosition_OneISO2Short,
+			},
+			senderCollateralPoolBalance:    big.NewInt(500),
+			recipientCollateralPoolBalance: big.NewInt(600),
+			senderCollateralPoolAddr: authtypes.NewModuleAddress(
+				types.ModuleName + ":" + lib.UintToString(constants.PerpetualPosition_OneISOShort.PerpetualId),
+			),
+			recipientCollateralPoolAddr: authtypes.NewModuleAddress(
+				types.ModuleName + ":" + lib.UintToString(constants.PerpetualPosition_OneISO2Short.PerpetualId),
+			),
+			quantums:    big.NewInt(500),
+			expectedErr: types.ErrFailedToUpdateSubaccounts,
+		},
+		"Send from isolated subaccount to non-isolated subaccount, sender does not have enough balance": {
+			asset:                *constants.Usdc,
+			senderAssetPositions: keepertest.CreateUsdcAssetPosition(big.NewInt(100)),
+			senderPerpetualPositions: []*types.PerpetualPosition{
+				&constants.PerpetualPosition_OneISOShort,
+			},
+			recipientAssetPositions: keepertest.CreateUsdcAssetPosition(big.NewInt(600)),
+			recipientPerpetualPositions: []*types.PerpetualPosition{
+				&constants.PerpetualPosition_OneBTCShort,
+			},
+			senderCollateralPoolBalance:    big.NewInt(500),
+			recipientCollateralPoolBalance: big.NewInt(600),
+			senderCollateralPoolAddr: authtypes.NewModuleAddress(
+				types.ModuleName + ":" + lib.UintToString(constants.PerpetualPosition_OneISOShort.PerpetualId),
+			),
+			recipientCollateralPoolAddr: types.ModuleAddress,
+			quantums:                    big.NewInt(500),
+			expectedErr:                 types.ErrFailedToUpdateSubaccounts,
+		},
+		"Send from non-isolated subaccount to isolated subaccount, collateral pool does not have enough balance": {
```

Comment:
for my understanding - collateral pools should never not have enough balance, right? should we panic instead of returning an error?

Reply:
Collateral pools could end up with not enough balance, in the same way the `subaccounts` module account could end up with not enough balance if withdrawal gating doesn't catch a negative TNC subaccount in time.
In terms of panic'ing, I think that could be fine, however it would diverge from the withdrawal flow, which returns an error.